### ISSUE TITLE
CRD Generic Sync broken in 0.19.0-beta.1

### ISSUE THREAD

**Issue by paalkr** [loft-sh/vcluster#1527]
### What happened?

I have this syncer config that looks like below. In vCluster `0.19.0-beta.0`, the `configuration`, `provider `and `context `CRDs are imported to the vCluster as expected. In vCluster vCluster `0.19.0-beta.1` importing the CRDs does not work, and I get this error message

```
2024-02-07 10:03:08     INFO    leaderelection/leaderelection.go:78     error creating configurations(terraform.appvia.io/v1alpha1) syncer: seems like resource terraform.appvia.io/v1alpha1, Kind=configurations is not available in the physical cluster or vcluster has no access to it  {"component": "vcluster"}
```

```
multiNamespaceMode:
  enabled: true
sync:
  generic:
    role:
      extraRules:
        - apiGroups: ["terraform.appvia.io"]
          resources: ["configurations"]
          verbs: ["create", "delete", "patch", "update", "get", "list", "watch"]

    clusterRole:
      extraRules:
        - apiGroups: ["apiextensions.k8s.io"]
          resources: ["customresourcedefinitions"]
          verbs: ["get", "list", "watch"]
        - apiGroups: ["terraform.appvia.io"]
          resources: ["contexts","providers"]
          verbs: ["get", "list", "watch"]
    config: |-
      version: v1beta1
      export:
        - apiVersion: terraform.appvia.io/v1alpha1
          kind: Configuration
      import:
        - apiVersion: terraform.appvia.io/v1alpha1
          kind: Provider
        - apiVersion: terraform.appvia.io/v1alpha1
          kind: Context
```

### What did you expect to happen?

The following CRDs to be present in the vCluster.
```
configurations.terraform.appvia.io
providers.terraform.appvia.io
contexts.terraform.appvia.io
```

### How can we reproduce it (as minimally and precisely as possible)?

Make a generic sync config that imports any CRD from the host cluster to the vCluster

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
v1.28.4-eks-8cb36c9
```

</details>

### Host cluster Kubernetes distribution

<details>

```
EKS
```

</details>

### vlcuster version

<details>

```console
0.19.0-beta.1
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k3s: v1.28.5+k3s1
```

</details>

### OS and Arch

<details>

```
OS: Amazon Linux 2
Arch: AMD64
```
</details>

**Comment by rohantmp** [comment]
Are you using the exact same physical clusters to test these versions? can you do `kubectl get crd` on the physical cluster where you reproduced this?

**Comment by paalkr** [comment]
Yes, I'm using the exact same host cluster.

**Comment by paalkr** [comment]
Seems to be fixed with 0.19.0 final

**Comment by rohantmp** [comment]
Ok! Thanks for updating, was just trying to repro :)
